gnucash/register/register-gnome/completioncell-gnome.c

```diff
@@ -0,0 +1,1010 @@
+
+            if (!string)
+                break;
+
+            g_signal_emit_by_name (G_OBJECT(box->item_list), "change_item",
+                                   string, (gpointer)bcell);
+
+            g_free (string);
+            break;
+        }
+    }
+
+    if (box->strict)
+        box->in_list_select = gnc_item_in_list (box->item_list, bcell->value);
+
+    if (!bcell->value)
+        item_store_clear (cell);
+
+    return FALSE;
+}
+
+void
+gnc_completion_cell_reverse_sort (CompletionCell* cell, gboolean is_reversed)
+{
+    PopBox* box;
+
+    if (cell == NULL)
+        return;
+
+    box = cell->cell.gui_private;
+
+    if (is_reversed != box->register_is_reversed)
+    {
+        gnc_completion_cell_clear_menu (cell);
+
+        box->register_is_reversed = is_reversed;
+        box->occurrence_id = 0;
+    }
+}
+
+static void
+toggle_filter_any_cb (GtkWidget *widget, gpointer user_data)
+{
+    PopBox* box = user_data;
+    box->filter_at_start = !gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM(widget));
+}
+
+static void
+toggle_use_alpha_cb (GtkWidget *widget, gpointer user_data)
+{
+    PopBox* box = user_data;
+
+    box->filter_sort_use_alpha = gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM(widget));
+
+    if (box->filter_sort_use_alpha)
+        box->sort_column_id = TEXT_NORMAL_COL;
+    else
+        box->sort_column_id = OCCURANCE_COL;
+}
+
+static void
+disconnect_secondary_icon (PopBox* box)
+{
+    if (box->icon_release_id != 0)
+        g_signal_handler_disconnect (G_OBJECT(box->item_edit->editor),
+                                     box->icon_release_id);
+    box->icon_release_id = 0;
+
+    gtk_entry_set_icon_from_icon_name (GTK_ENTRY(box->item_edit->editor),
+                                       GTK_ENTRY_ICON_SECONDARY,
+                                       NULL);
+}
+
+static void
+icon_release_cb (GtkEntry *entry, GtkEntryIconPosition icon_pos,
+                 GdkEvent* event, gpointer user_data)
+{
+    PopBox* box = user_data;
+    GtkWidget *menu, *filter_any, *use_alpha = NULL, *msg_label;
+
+    if (icon_pos != GTK_ENTRY_ICON_SECONDARY)
+        return;
+
+    menu = gtk_menu_new ();
+    filter_any = gtk_check_menu_item_new_with_mnemonic (_("Filter any where"));
+    gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM(filter_any), !box->filter_at_start);
+//    gtk_widget_set_tooltip_text (filter_any, _("Popup after 2 characters"));
+
+    if (box->sort_enabled)
+    {
+        use_alpha = gtk_check_menu_item_new_with_mnemonic (_("Normal filter sorting"));
+        gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM(use_alpha), box->filter_sort_use_alpha);
+        g_signal_connect (G_OBJECT(use_alpha), "toggled",
+                          G_CALLBACK(toggle_use_alpha_cb), box);
+    }
+    gchar *msg = g_strdup_printf ("%d %s", g_hash_table_size (box->item_hash), _("Unique Entries"));
+    msg_label = gtk_menu_item_new_with_label (msg);
```

Comment:
This is debug code. Although perhaps the menu will not stick, remember to remove it for the final implementation.

Reply:
Originally I was going to leave that but not required as the secondary icon is scrapped.